### PR TITLE
Qqsuasar dlas

### PR DESCRIPTION
**Pull request by alxogm**
@julienguy New try to merge with master. Includes dla and bal features

### CODE REVIEW

File: py/desisim/scripts/quickquasars.py [hunk 1]
```diff
@@ -245,7 +283,12 @@ def simulate_one_healpix(ifilename,args,model,obsconditions,decam_and_wise_filte
         nmodel=nqso, redshift=metadata['Z'], 
         lyaforest=False, nocolorcuts=True, noresample=True, seed = seed)
 
+    ##To add BALs to be checked by Luz and Jaime
+    if args.balprob:
+       bal=BAL()
+       tmp_qso_flux,meta_bal=bal.insert_bals(tmp_qso_wave,tmp_qso_flux, metadata['Z'], balprob=args.balprob)
```
Comment (moustakas) [inline py/desisim/scripts/quickquasars.py]: You should pass `seed=seed` to `bal.insert_bals()` on line 289 to ensure reproducibility (i.e., that the same QSOs become BAL QSOs when the seed is the same).

Also consider passing the `BAL()` object as an argument to `simulate_one_healpix` so you don't read the BAL basis templates over and over.

**Comment by alxogm** [comment]
@moustakas I added the seed, but passing the BAL() as argument, maybe I should leave it for @forero who is going to take over the BALs simulations in general, also note I'm not saving the BAL metadata yet.

**Comment by alxogm** [comment]
@julienguy I've checked that not dlas are being added if --dla is not present... in line 154 I only read the skewers from transmission file the there are no dlas or they are random...

**Comment by alxogm** [comment]
@julienguy  I just fixed to add BALs only when --balprob is present, and since I was checking this piece the  I included the suggestion from @moustakas  to pass BAL() as argument to simulate_one_pixel... 

Could you check if there are more requests?

**Pull request merged**